File: tests/cs-sdk/unit/nogui_signal_subscriber_test.cs
// unit test: SignalSubscriber (nogui)
using System;
using System.Threading;
using System.Threading.Tasks;
using KnotLink;

var sub = new SignalSubscriber("app.knotlinksdk.test", "test_signal")
{
    OnSignalAsync = data =>
    {
        Console.WriteLine($"Received: {data}");
        return Task.CompletedTask;
    }
};
Console.WriteLine("Subscriber listening...");
Thread.Sleep(60000);
sub.Dispose();

File: knotlink-sdk-cpp/tests/unit/nogui_signal_subscriber_test.cpp
// unit test: SignalSubscriber (nogui)
#include "../../knotlink/SignalSubscriber.hpp"
#include <iostream>
#include <thread>
#include <chrono>
using namespace knotlink;

int main() {
    SignalSubscriber sub("app.knotlinksdk.test", "test_signal");
    sub.setOnDataReceivedCallback([](const std::string& data) {
        std::cout << "Received: " << data << std::endl;
    });
    std::cout << "Subscriber listening..." << std::endl;
    std::this_thread::sleep_for(std::chrono::seconds(60));
    return 0;
}
